build/Makefile

```
@@ -0,0 +1,167 @@
+YQ_INSTALL_PATH:=./yq
+YQ_PATH:=$(shell which yq || echo $(YQ_INSTALL_PATH))
+YQ_INSTALL_COMMAND:=curl $(CURL_INSTALL_OPTS) -o $(YQ_INSTALL_PATH) $(YQ_INSTALL_URL) && chmod +x $(YQ_INSTALL_PATH)
+
+$(YQ_PATH):
+	@$(YQ_INSTALL_COMMAND)
+
+$(XMLLINT_PATH):
+	@$(XMLLINT_INSTALL_COMMAND)
+
+.PHONY: dependencies
+dependencies: $(JQ_PATH) $(XMLLINT_PATH) $(YQ_PATH) ## Install binary build jq and yq dependencies for repo
+
+.PHONY: artifacts
+artifacts: copy-readmes copy-xml-content resolve-xml-profiles convert-min-json-content reformat-json-content convert-yaml-content ## Generate all artifacts
+
+.PHONY: copy-readmes
+copy-readmes: $(GEN_READMES) ## Copy README files to release location
+
+$(GEN_CONTENT_DIR)/%.md: $(SRC_DIR)/%.md
+	# $(@D): The directory part of the file name of the target, with the
+	# trailing slash removed.
+	# https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html
+	@mkdir -p $(@D)
+	@cp $(SRC_DIR)/$*.md $(GEN_CONTENT_DIR)/$*.md
+
+.PHONY: copy-xml-content
+copy-xml-content: $(GEN_XML_FILES) ## Copy OSCAL XML files to release location
+
+$(GEN_CONTENT_DIR)/%.xml: $(SRC_DIR)/%.xml
+	@mkdir -p $(@D)
+	@cp $(SRC_DIR)/$*.xml $(GEN_CONTENT_DIR)/$*.xml
+
+.PHONY: resolve-xml-profiles
+resolve-xml-profiles: $(GEN_XML_RESOLVED_CATALOGS) ## Resolve OSCAL XML profiles for custom catalogs
+
+$(GEN_CONTENT_DIR)/%-resolved-profile_catalog.xml: $(SRC_DIR)/%_profile.xml
+	mkdir -p $(@D)
+	$(PROFILE_RESOLVER_RUNNER) $(GEN_CONTENT_DIR)/$*_profile.xml $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml $(PROFILE_RESOLVER_ARGS)
+	@sed -i'' -e "s|file:$(shell realpath $(GEN_CONTENT_DIR)/$*_profile.xml)|$(shell basename $*_profile.xml)|g" $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml	
+
+.PHONY: validate-xml-content
+validate-xml-content: $(GEN_XML_FILES) ## Validate XML files
+	@xmllint --schema $(OSCAL_COMPLETE_XML_SCHEMA) --noout $(GEN_XML_FILES)
+
+.PHONY: convert-min-json-content
+convert-min-json-content: $(GEN_MIN_JSON_FILES) ## Convert examples from OSCAL XML to JSON
+
+.SECONDEXPANSION:
+$(GEN_CONTENT_DIR)/%-min.json: $(GEN_CONTENT_DIR)/$$(subst json,xml,%).xml
+	@mkdir -p $(@D)
+	$(XSLT_RUNNER) $(XML_JSON_CONVERTER_XSLT) $(GEN_CONTENT_DIR)/$(subst json,xml,$*).xml $(GEN_CONTENT_DIR)/$*-min.json
+
+.PHONY: reformat-json-content
+reformat-json-content: $(GEN_JSON_FILES) ## Format minified JSON to pretty-printed JSON
+
+$(NPM_PREFIX_DIR)/$(NPM_PKGS_DIR):
+	$(MAKE) -C oscal/build dependencies
+
+$(GEN_CONTENT_DIR)/%.json: $(GEN_CONTENT_DIR)/%-min.json
+	$(JQ_PATH) . $(GEN_CONTENT_DIR)/$*-min.json > $(GEN_CONTENT_DIR)/$*.json
+
+.PHONY: checks
+checks: validate-xml-content validate-json-content validate-yaml-content ## Check all content with schema and other validation methods
+
+.PHONY: validate-json-content
+validate-json-content: $(GEN_JSON_FILES) ## Validate JSON files
+	npx --prefix $(NPM_PREFIX_DIR) ajv validate -s $(OSCAL_COMPLETE_JSON_SCHEMA) -c ajv-formats $(foreach file,$(GEN_JSON_FILES),-d $(file))
```

B: NPM install dependancy
```suggestion
validate-json-content: $(GEN_JSON_FILES) $(NPM_PREFIX_DIR)/$(NPM_PKGS_DIR) ## Validate JSON files
	npx --prefix $(NPM_PREFIX_DIR) ajv validate -s $(OSCAL_COMPLETE_JSON_SCHEMA) -c ajv-formats $(foreach file,$(GEN_JSON_FILES),-d $(file))
```

---

Corrected in https://github.com/usnistgov/oscal-content/commit/99669b4bd2603f0eea88a5d8a2ca2df76921b5d6.